### ISSUE TITLE
Factor indexer support out of maptos-opt-executor

### ISSUE THREAD

**Issue by mzabaluev** [movementlabsxyz/movement#241]
**Is your feature request related to a problem? Please describe.**
The support for the indexer service was added to maptos-opt-executor.
This adds significantly to the dependencies, including some native dependencies like libpq, which may be undesirable for SDK crate users that need just the executor protocol unit.

**Describe the solution you'd like**
Consider moving the indexer service into a separate micro-crate with its own dependencies.

**Describe alternatives you've considered**
If not possible to make a clean split, consider feature-gating the indexer support.

**Additional context**
`cargo tree -p maptos-opt-executor` before and after #210 got merged into main.

The current implementation of the indexer shares the API context, the database connection, and bits of configuration with the executor.

**Comment by musitdev** [comment]
The indexer service has been removed from Suzuka executor and moved to a separate service that use gRpc indexer entry point of the Suzuka node to get notified of new transactions.
See PR #295 #431  #436

**Comment by musitdev** [comment]
Need an e2e test in production env with current Suzuka node Testnet DB to validate the initial dump of the indexer DB.
